### PR TITLE
Sem-Ver: feature Add support for using PyJWT 2.0.x

### PR DESCRIPTION
**Pull request by dbaxa**
These changes provide support for using PyJWT 1.x and 2.0.x.

### CODE REVIEW

File: atlassian_jwt_auth/signer.py [hunk 1]
```diff
@@ -72,11 +72,14 @@ def generate_jwt(self, audience, **kwargs):
...
             key_identifier, private_key_pem)
-        return jwt.encode(
+        token = jwt.encode(
             self._generate_claims(audience, **kwargs),
             key=private_key,
             algorithm=self.algorithm,
             headers={'kid': key_identifier.key_id})
+        if isinstance(token, str):
+            token = token.encode('utf-8')
+        return token
```
Comment (dbaxa) [inline atlassian_jwt_auth/signer.py:82]: To remain consistent.

**Comment by dbaxa** [comment]
Related: https://github.com/jpadilla/pyjwt/blob/master/CHANGELOG.rst

File: atlassian_jwt_auth/signer.py (hunk 1 above)
Comment (jbirch-atlassian) [reply atlassian_jwt_auth/signer.py:82]: Ah really, PyJWT changed between bytes and str between versions?

OOC, if and when support only PyJWT 2+, will we also change this to be in line with the types it returns?

**Review by jbirch-atlassian — APPROVED** [review]
Seems cool to me. Sorta annoying we have to repeat ourselves to be compatible with both versions — is there any particular reason we didn't just bump to PyJWT 2+?

File: atlassian_jwt_auth/signer.py (hunk 1 above)
Comment (dbaxa) [reply atlassian_jwt_auth/signer.py:82]: Possibly, although it might result in un-needed churn :shrug: .

**Comment by dbaxa** [comment]
> Seems cool to me. Sorta annoying we have to repeat ourselves to be compatible with both versions — is there any particular reason we didn't just bump to PyJWT 2+?

For compatibility reasons and to not need a major version bump so soon after performing a major version bump :-) .

**Pull request merged**